src/System.Console/tests/ReadAndWrite.cs

```diff
@@ -261,6 +259,30 @@ public static unsafe void ValidateConsoleEncoding(Encoding encoding)
     }
 
     [Fact]
+    // On the full framework it is not guaranteed to eat the preamble bytes
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+    public static unsafe void OutputEncodingPreamble()
+    {
+        Encoding curEncoding = Console.OutputEncoding;
+
+        try
+        {
+            Encoding encoding = Console.Out.Encoding;
+            // The primary purpose of ConsoleEncoding is to return an empty preamble.
+            Assert.Equal(Array.Empty<byte>(), encoding.GetPreamble());
+
+            // Try setting the ConsoleEncoding to something else and see if it works.
+            Console.OutputEncoding = Encoding.Unicode;
+            // The primary purpose of ConsoleEncoding is to return an empty preamble.
+            Assert.Equal(Array.Empty<byte>(), encoding.GetPreamble());
```

Comment:
This is still checking ```encoding```, which hasn't been modified since the earlier same check a few lines earlier... what am I missing?